docs/framework/data/adonet/retrieving-data-using-a-datareader.md

```diff
@@ -1,131 +1,140 @@
-  
-while (custReader.Read())  
-{  
-  Console.WriteLine("Orders for " + custReader.GetString(1));   
-  // custReader.GetString(1) = CompanyName  
-  
-  orderReader = (OleDbDataReader)custReader.GetValue(2);        
-  // custReader.GetValue(2) = Orders chapter as DataReader  
-  
-  while (orderReader.Read())  
-    Console.WriteLine("\t" + orderReader.GetInt32(1));          
-    // orderReader.GetInt32(1) = OrderID  
-  orderReader.Close();  
-}  
-// Make sure to always close readers and connections.  
-custReader.Close();  
-}  
+using (OleDbConnection connection = new OleDbConnection(
+    "Provider=MSDataShape;Data Provider=SQLOLEDB;" +
+    "Data Source=localhost;Integrated Security=SSPI;Initial Catalog=northwind"))
+{
+    using (OleDbCommand custCMD = new OleDbCommand(
+        "SHAPE {SELECT CustomerID, CompanyName FROM Customers} " +
+        "APPEND ({SELECT CustomerID, OrderID FROM Orders} AS CustomerOrders " +
+        "RELATE CustomerID TO CustomerID)", connection))
+    {
+        connection.Open();
+
+        using (OleDbDataReader custReader = custCMD.ExecuteReader())
+        {
+
+            while (custReader.Read())
+            {
+                Console.WriteLine("Orders for " + custReader.GetString(1));
+                // custReader.GetString(1) = CompanyName  
+
+                using (OleDbDataReader orderReader = (OleDbDataReader)custReader.GetValue(2))
+                {
+                    // custReader.GetValue(2) = Orders chapter as DataReader  
+
+                    while (orderReader.Read())
+                        Console.WriteLine("\t" + orderReader.GetInt32(1));
+                    // orderReader.GetInt32(1) = OrderID  
+                    orderReader.Close();
+                }
+            }
+            // Make sure to always close readers and connections.  
+            custReader.Close();
+        }
+    }
+}
 ```  
   
 ## Returning Results with Oracle REF CURSORs  
  The .NET Framework Data Provider for Oracle supports the use of Oracle REF CURSORs to return a query result. An Oracle REF CURSOR is returned as an <xref:System.Data.OracleClient.OracleDataReader>.  
   
- You can retrieve an **OracleDataReader** object, that represents an Oracle REF CURSOR using the <xref:System.Data.OracleClient.OracleCommand.ExecuteReader%2A> method, and you can also specify an <xref:System.Data.OracleClient.OracleCommand> that returns one or more Oracle REF CURSORs as the **SelectCommand** for an <xref:System.Data.OracleClient.OracleDataAdapter> used to fill a <xref:System.Data.DataSet>.  
+ You can retrieve an **OracleDataReader** object that represents an Oracle REF CURSOR by using the <xref:System.Data.OracleClient.OracleCommand.ExecuteReader%2A> method. You can also specify an <xref:System.Data.OracleClient.OracleCommand> that returns one or more Oracle REF CURSORs as the **SelectCommand** for an <xref:System.Data.OracleClient.OracleDataAdapter> used to fill a <xref:System.Data.DataSet>.  
```

Comment:
OracleDataReader can be an xref (\<xref:System.Data.OracleClient.OracleDataReader>)